src/ast/passes/codegen_llvm.cpp

```diff
@@ -2367,27 +2367,32 @@ void CodegenLLVM::visit(AssignMapStatement &assignment)
     b_.CreateLifetimeEnd(val);
 }
 
-void CodegenLLVM::visit(AssignVarStatement &assignment)
+void CodegenLLVM::updateVariableType(const std::string &var_ident,
+                                     const SizedType &var_type)
 {
-  Variable &var = *assignment.var;
-
-  auto scoped_del = accept(assignment.expr);
-
-  if (variables_.find(var.ident) == variables_.end()) {
-    SizedType alloca_type = var.type;
-
+  if (variables_.find(var_ident) == variables_.end()) {
```

Comment:
Nit: inverse the condition and return early to reduce nesting
```suggestion
  if (variables_.find(var_ident) != variables_.end())
    return
```